scripts/ctest_parser.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+"""
+ Copyright (C) 2022 Intel Corporation
+
+ Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ See LICENSE.TXT
+ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+from sys import argv
+from subprocess import PIPE, Popen, STDOUT
+import argparse
+import os
+
+
+def run(command, env):
```

Comment:
gtest can be made to output json, which is likely to be more reliable than parsing text.

Reply:
Didn't realise that, sounds like it could be better. The script works by running `ctest` in `build/test/conformance` rather than `gtest` directly against the the executables in `build/bin` which had the advantage of not needing to maintain a list of tests to run against. Although it was possible to first parse the relevant executables paths contained in each of the `CTestTestfile.cmake` in `build/test/conformance/<component>` so a list of relevant tests is built automatically.

I'll take a look at the json output from gtest, thanks.

Reply:
Might be worth checking if there's an environment variable for gtest that will enable the json output without modifying the test commands in the cmake?

Reply:
I was looking into this briefly. `gtest` has a ``--gtest_output=json:<outfile>`` which includes all the information about passed tests etc. I'm not sure however if you can forward this argument through Ctest into gtest or if you have to manually call each gtest executable.

Reply:
> Although we could modify the CMake to invoke gtest with this argument i.e.
> 
> ```cmake
>     add_test(NAME ${name}
>         COMMAND ${TEST_TARGET_NAME} "--gtest_output=json:${CMAKE_CURRENT_BINARY_DIR}/${name}-output.json"
>         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
> ```

That makes the most sense to me. Summarizing results would then become a post-processing step on `*-output.json` files. This approach also has the added benefit of no longer requiring python to launch ctests directly.

Reply:
A problem with asking for the json output is if a test crashes (e.g. a seg fault is produced) then the suite and executable will also crash, and no output file gets produced.

Reply:
That's correct, you will only get details from the tests prior to the crash, anything after is lost. With my script I first check how many tests are part of the suite, giving some idea of a pass rate but yes it wouldn't be 100% accurate as there may be a passing test blocked behind an earlier crashed test.

I was thinking a way around that is to run `gtest --gtest-list-tests` to get the names of all tests, then run them individually, but many tests rely on objects being created in the suite init functions.

Reply:
If we expect crashing tests to be a regular occurrence, I think it might be better to wrap all tests in `EXPECT_EXIT` to run each one as a separate process.